src/data/reddit_dataset.py

```diff
@@ -65,22 +64,22 @@ def __iter__(self) -> Iterator[tuple[str, str]]:
             with open(file, "rt") as f_in:
                 for line in f_in:
                     sample = json.loads(line)
-                    utterances = [self.bos_token + it + self.eos_token for it in sample["thread"]]
+                    utterances = sample["thread"]
 
                     if not self.multiple_samples:  # yield only one sample per thread
                         if self.single_turn:  # yield only 1 utterance per context (use 0th)
-                            yield utterances[0], utterances[1]
+                            yield self.preprocess(utterances[0], utterances[1])
                             continue
                         # yield full thread
-                        yield " ".join(utterances[:-1]), utterances[-1]
+                        yield self.preprocess(utterances[:-1], utterances[-1])
                         continue
 
                     for i in range(1, len(utterances)):
                         if self.single_turn:  # yield previous utterance as context
-                            yield utterances[-1], utterances[i]
+                            yield self.preprocess(utterances[-1], utterances[i])
```

Comment:
This is probably my typo: the previous utterance should have index `i-1`, not `-1`. Could you fix it?